### ISSUE TITLE
warning unable to download

### ISSUE THREAD

**Issue by delanym** [openrewrite/rewrite-maven-plugin#264]
Split off #133 

I get warnings for not being able to download artifacts that are **both** in the ~/.m2/repository and in a local Maven repository

```xml
<repositories>
  <repository>
    <id>thirdparty</id>
    <name>Third-party Artefacts</name>
    <url>file:///tmp/provided</url>
  </repository>
</repositories>
```

**Comment by tkvangorder** [comment]
@delanym Thanks for reporting the issue, we really need a little more information to correctly triage this issue. Are you saying that if an artifact exists in a repository that is defined via a local file path, that those artifacts are not being resolved? Or are you saying that when you add a local file repository that even artifacts in the local repository cache are no longer found?

**Comment by delanym** [comment]
@tkvangorder yes to first question. As to second, I can't answer that since I never added a local file repository - it was always there. Its complaining about artefacts of other modules that I know to be in the cache.

Here some instructions for you to get the warning for a missing izpack dependency. If the dependency is truly invalid it would fail before even getting to the rewrite plugin.

```
git clone https://github.com/delanym/buildplan-maven-plugin.git
cd build-maven-plugin
./mvnw validate -Drewrite
```

Why should I get warnings for missing dependencies during "Parsing Java main files..."
Rewrite plugin is clearly stepping over Maven's dependency mechanism.

**Comment by aegershman** [comment]
@delanym Currently looking into this

**Comment by rdifrango** [comment]
I running into the same thing with Maven; but we're using Artifactory internally as the proxy for Maven central.

The application compiles and runs fine as it can find the dependencies.

**Comment by aegershman** [comment]
@delanym I'm going to move this issue to the main `rewrite` repository. The root of this appears to be in MavenPomDownloader. Here we're building a request to the 'remote' repository using OkHttp and passing it a URI of `file://xyz`. OkHttp then throws an IllegalArgument exception as it's expecting an `http(s)` protocol.

https://github.com/openrewrite/rewrite/blob/111fdccef63855e1387b8591727110fc2759ce9b/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L361-L362

@rdifrango that sounds like something unexpected when using a Maven mirror? If so, could you log a separate issue about it with a sample of what you're seeing and what you have configured? I'd like to keep this issue specifically dedicated to `<repository>`'s defined with a local filesystem path 👍
